rusk/src/lib/services/state.rs

```diff
@@ -173,6 +173,7 @@ impl State for Rusk {
         info!("Received ExecuteStateTransition request");
 
         let mut state = self.state()?;
+        let mut forked_state = state.fork();
```

Comment:
This can be optimised, forking only if the tx.GasLimit is greater than the block.gasLeft

Reply:
This would be possible, but would introduce a branch that would increase the code footprint substantially for marginal gains. Lets tackle this in the future.